modules/http/src/main/java/org/mule/module/http/internal/listener/HttpResponseBuilder.java

```diff
@@ -113,7 +116,14 @@ public HttpResponse build(org.mule.module.http.internal.domain.response.HttpResp
             final Collection<String> paramValues = resolvedHeaders.getAll(name);
             for (String value : paramValues)
             {
-                httpResponseHeaderBuilder.addHeader(name, value);
+                if (TRANSFER_ENCODING.equals(name) && !supportsTransferEncoding(event))
+                {
+                    logger.debug("Client HTTP version is lower than 1.1 so the unsupported 'Transfer-Encoding' header has been removed and 'Content-Length' will be sent instead.");
```

Comment:
Add "if (logger.isDebugEnabled) ". (I know there are other calls to logger.debug in this class without this check, but they are wrong)